### DISCUSSION TITLE
Why can't my code pass the baggage content downstream?

### DISCUSSION THREAD

**Discussion by shenlvcheng** [open-telemetry/opentelemetry-java-instrumentation#10403] [Q&A]
The upstream code is as follows
`Context context = Context.current();
		Baggage baggage = Baggage.fromContextOrNull(context);
		if (baggage == null) {

			baggage = Baggage.builder().put("key","valye").put("key1","sharif").build();

		} else {
			baggage.toBuilder().put("key","value").build();
		}
		baggage.storeInContext(context);`

The downstream cannot retrieve the baggage; the downstream code is as follows. 
``Context context = Context.current();
		Baggage baggage = Baggage.fromContextOrNull(context);
		if (baggage == null) {

			baggage = Baggage.builder().put("key","valye").put("key1","sharif").build();

		} else {
			baggage.toBuilder().put("key","value").build();
		}
		baggage.storeInContext(context);``
The downstream code for baggage returns null:
`Baggage baggage = Baggage.fromContextOrNull(context);`

Using HttpClient 4.0 as the framework for HTTP requests to interact.

Using OpenTelemetry Java Agent 2.0.

**Comment by trask** [comment]
hi @shenlvcheng! `baggage.storeInContext(context)` returns a new `Context` object which you need to `makeCurrent()`

**Reply by zeitlinger** [reply]
Here's a [working example](https://github.com/open-telemetry/opentelemetry-java-examples/blob/1a4c8cf49be597ee97f7470b18fe8b72181eba26/manual-tracing/src/main/java/io/opentelemetry/example/tracing/ManualTracingExample.java#L33-L44)
